Adding orvara eccentric anomaly solver and the system RV calculations based on it

CoreySpohn opened the pull request:
Changes:
- Addition of an "RV_C" directory that hosts the code necessary for the RV calculations
- Modifications of the "eccanom_C" directory files to implement the solver from orvara
- Modifications of the "fun.py" file to implement the functions: `eccanom_orvara`, `calc_RV_from_M`, and `calc_RV_from_time`
- I changed the install to require cython so that the C extensions get installed and a pyproject.toml file with the build system requirements for cython/numpy. I can easily revert that change if you have use cases that force you not to have Cython installed.

One thing to note is that because the orvara E solver was built in conjunction with a fast RV calculation the `eccanom_orvara`, by default, returns the sine and cosine of E. I didn't modify the behavior of that, since the RV calculation relies on it, but I think it makes sense to give the base `eccanom` function an input to use the orvara solver (like "use_orvara=True" or "method='orvara'") and add some logic inside `eccanom` that calls `eccanom_orvara` but only returns E. However, I don't know what you'd want to do to implement it.

All code was formatted with black and I tried to keep the docstring/comment style consistent with the existing code. Tests ran successfully on my machine in a clean virtualenv. There shouldn't be any changes to the functions that already exist.

CoreySpohn committed:
debug file

CoreySpohn committed:
Adding orvara's solver

CoreySpohn committed:
Support for RV calculations

CoreySpohn committed:
Fixing header file

CoreySpohn committed:
Updating with better comments and formatting, tested the RV numbers more
carefully

CoreySpohn committed:
Merge branch 'rv_calc' of github.com:CoreySpohn/keplertools into rv_calc

CoreySpohn committed:
removing test file

CoreySpohn committed:
removing pyproject

CoreySpohn committed:
Updating install and adding gitignore for the CyRV.c file

dsavransky commented on setup.py:
```diff
@@ -42,17 +58,15 @@
-    },
+    packages=setuptools.find_packages(exclude=["tests*"]),
+    install_requires=["numpy", "cython"],
+    extras_require={"C": ["cython"]},
```
I'm fine with making cython an absolute requirement.  However, this makes the 'C' extra redundant.  Can you please remove it altogether?  Thanks.

CoreySpohn replied:
Makes sense, just pushed that change

CoreySpohn committed:
Removing the extras_require flag from setup.py

dsavransky commented:
Sorry, one more request: is it possible to recycle some of the existing unit tests to apply to the new stuff as well?

CoreySpohn committed:
Adding tests for eccanom_orvara function

Comparisons between the regular eccanom and eccanom_orvara, fixing handling of scalar M value

CoreySpohn committed:
Relaxing comparison, since orvara only guarantees precision to 10^-14 at high eccentricity

CoreySpohn commented:
I added a comparison of the base `E` (from `eccanom`) against `E_orvara` (from `eccanom_orvara`) in the tests of the two input combinations that `eccanom_orvara` is built to support (scalar/vector M + scalar e). I didn't set `eccanom_orvara` up to work with a vector of eccentricity values, since the speed-up comes from pre-computing terms to get good initial guesses for E for a single e. I'd be surprised if orvara is worth using for single e/M pairs.
